### DISCUSSION TITLE
Email invite / verification not send

### DISCUSSION THREAD

**Discussion by LelieL91** [dani-garcia/vaultwarden#5129] [General]
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.32.2-r0
* Web-vault version: v2024.6.2c
* OS/Arch: linux/x86_64
* Running within a container: false (Base: Not applicable)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Forwarded-For)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.46.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** ADMIN_TOKEN

```json
{
  "_duo_akey": null,
  "_enable_duo": true,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "*****://***********",
  "attachments_folder": "/var/lib/vaultwarden/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "/var/lib/vaultwarden",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "*******************************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*****************",
  "domain_origin": "*****://*****************",
  "domain_path": "",
  "domain_set": true,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "enforce_single_org_with_reset_pw_policy": false,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": "****************",
  "hibp_api_key": null,
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "/var/lib/vaultwarden/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Raspify Vault",
  "invitations_allowed": true,
  "ip_header": "X-Forwarded-For",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "/var/lib/vaultwarden/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "/var/lib/vaultwarden/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": "Login",
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "********************",
  "smtp_from_name": "Raspify Vault",
  "smtp_host": "*****************",
  "smtp_password": "***",
  "smtp_port": 465,
  "smtp_security": "force_tls",
  "smtp_ssl": null,
  "smtp_timeout": 60,
  "smtp_username": "********************",
  "templates_folder": "/var/lib/vaultwarden/templates",
  "tmp_folder": "/var/lib/vaultwarden/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "/usr/share/webapps/vaultwarden-web/",
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

### Vaultwarden Build Version

v1.32.2-r0

### Deployment method

OS Package (apt, yum/dnf, pacman, apk, nix, ...)

### Reverse Proxy

nginx 1.27.2-1

### Host/Server Operating System

Linux

### Operating System Version

Alpine v3.20

### Clients

Web Vault

### Client Version

Firefox / Opera Latest

### Steps To Reproduce

1. Invite a new user
2. After new user login, email verification 

### Expected Result

Email received

### Actual Result

No email are received, while the test mail in admin section and the new login notification works as expected

### Logs

```text
/--------------------------------------------------------------------\
|                        Starting Vaultwarden                        |
|                         Version 1.32.2-r0                          |
|--------------------------------------------------------------------|
| This is an *unofficial* Bitwarden implementation, DO NOT use the   |
| official channels to report bugs/features, regardless of client.   |
| Send usage/configuration questions or feature requests to:         |
|   https://github.com/dani-garcia/vaultwarden/discussions or        |
|   https://vaultwarden.discourse.group/                             |
| Report suspected bugs/issues in the software itself at:            |
|   https://github.com/dani-garcia/vaultwarden/issues/new            |
\--------------------------------------------------------------------/

[INFO] Using saved config from `/var/lib/vaultwarden/config.json` for configuration.

[WARNING] The following environment variables are being overridden by the config.json file.
[WARNING] Please use the admin panel to make changes to them:
[WARNING] ADMIN_TOKEN

[2024-10-24 18:36:05.004][start][INFO] Rocket has launched from http://0.0.0.0:8000
[2024-10-24 18:37:01.531][request][INFO] GET /api/accounts/revision-date
[2024-10-24 18:37:01.532][response][INFO] (revision_date) GET /api/accounts/revision-date => 200 OK
[2024-10-24 18:37:02.717][request][INFO] POST /identity/connect/token
[2024-10-24 18:37:02.723][response][INFO] (login) POST /identity/connect/token => 200 OK
[2024-10-24 18:37:02.841][request][INFO] POST /api/accounts/verify-email
[2024-10-24 18:37:04.236][response][INFO] (post_verify_email) POST /api/accounts/verify-email => 200 OK
[2024-10-24 18:42:13.827][request][INFO] GET /api/config
[2024-10-24 18:42:13.828][response][INFO] (config) GET /api/config => 200 OK
[2024-10-24 18:42:47.871][request][INFO] GET /admin/diagnostics
[2024-10-24 18:42:49.441][response][INFO] (diagnostics) GET /admin/diagnostics => 200 OK
[2024-10-24 18:42:58.042][request][INFO] GET /admin/diagnostics/config
[2024-10-24 18:42:58.043][response][INFO] (get_diagnostics_config) GET /admin/diagnostics/config => 200 OK
```

### Screenshots or Videos

No errors on Web UI or log files

**Comment by BlackDex** [comment]
Please try to enable `SMTP_DEBUG=true` and see what happens.
Also, an invite via mail once used already validates the mail.

Since there are no errors, i do not believe something is wrong on the  Vaultwarden side.

**Reply by LelieL91** [reply]
> Please try to enable `SMTP_DEBUG=true` and see what happens. Also, an invite via mail once used already validates the mail.
> 
> Since there are no errors, i do not believe something is wrong on the Vaultwarden side.

Where should I enable this option? I don't use docker

I'm using vaultwarden for more than 1 year, never had a problem with it, I already got 2 verified account on it, but yesterday I tried adding a new account by invitation but it never arrived.
I instantly thought about the SMTP settings and the email server, but when I tried the "test email", that one arrived and also the login notification by email arrived as well. So the problem can't be the mail server or the Mailu SMTP settings or no email will be sent.

Thank you for your time and assistance

**Reply by BlackDex** [reply]
You should set an ENV variable like i posted above.
I'm not sure how you configured your application.
If the basic config is located in the `mailu.env` then yes. But you have to check/verify how the package you installed expects the basic config.

**Reply by LelieL91** [reply]
forget about the "mailu.env", I was thinking about the mail server. About the ENV var, I'm not using docker, I can't set this like in a compose

**Reply by BlackDex** [reply]
Again, i have no clue how your package loads it basic config, like where the web-vault is located etc.. Since there are non-editable config options on the Vaultwarden Admin Backend `/admin` which you need to configure via ENV variables.
So, i would suggest to check how that works with the package you installed, as those are not maintained by this project, i have no idea how that works.

**Reply by LelieL91** [reply]
I see the option as a read-only config, should I edit manually the file located in /var/lib/vaultwarden/config.json and add `"smpt_debug": true,`?
![immagine](https://github.com/user-attachments/assets/681077eb-5021-4d46-b9a9-022496631aae)

**Reply by LelieL91** [reply]
> Again, i have no clue how your package loads it basic config, like where the web-vault is located etc.. Since there are non-editable config options on the Vaultwarden Admin Backend `/admin` which you need to configure via ENV variables. So, i would suggest to check how that works with the package you installed, as those are not maintained by this project, i have no idea how that works.

alright, I'll do a couple of test and let you know

**Reply by BlackDex** [reply]
My guess would be, since you are using Alpine, you also installed the open-rc package, then you need to configure this in `/etc/conf.d/vaultwarden`

**Reply by LelieL91** [reply]
> My guess would be, since you are using Alpine, you also installed the open-rc package, then you need to configure this in `/etc/conf.d/vaultwarden`

You were right, sorry I feel very dumb today.
Btw I noticed that I was using the old conf, so replaced with the new one, edited all the custom export and enabled the SMPT_DEBUG.
There is no error in the log, everything seems fine. I did a test on an internal mail, and the invite arrived as usual, so the problem seems related to gmail filter I guess.
But I don't get why the test mail and the new login email notification are still arriving on gmail, but not invite or email verification.

**Reply by BlackDex** [reply]
Checked the spam box?
And, during those email's when sent, there isn't a warning message there maybe?
Sometimes a mail server rejects a first attempt and the sender needs to try again ( graylisting ), which is not what Vaultwarden does support (or will).

If Vaultwarden is able to deliver the mail without issues, then it is either your SMTP provider, or if you deliver it via GMail it self, something else might be the case.

So, check the Spam (And maybe even the sent) folders.

**Reply by LelieL91** [reply]
I'm already checking the spam box, the mails that got delivered (test and login notif) are there.
I have a selfhost mail server and I'm trying to invite a new user that got gmail. Since my home ip is kind blacklisted for email, I already confirmed that outlook is blocking me entirely, but on gmail I got my mails on spam box.
Since I already tested and received in spam box both manual mail and mailu test mail/login, I'm wondering if there is any other flag on invite/mail verification that makes gmail filter those.
Otherwise I can't understand why this double behaviour, using the same mail settings, test mails are fine but invite not.

**Reply by BlackDex** [reply]
Since we do not know how filtering works on GMail (Or Microsoft) there is probably no sane way in figuring this out.
Either use some other mail provider, or disable it all together, but the latter is probably not wanted.

**Reply by LelieL91** [reply]
I don't know why this strange behavior happens, as for my logic or I receive all mails or none, but if I use a smtp server with ip not listed as unsecure for spam, the verification mail/invite arrives.
Sorry for the issue and all the bother. Thank you for your time and your work

**Reply by BlackDex** [reply]
np, troubleshooting takes time.

**Comment by LelieL91** [comment]
Hi,
since I open this issue, I saw multiple commits regarding SMTP/invitation, I'm currently running v1.33.1 (latest not yet updated on alpine repo) and I just tried again sending the test mail and the invitation name to the same address (gmail), and it seems now working as expected, even if the destination mail marks those as spam, they both arrives (not only the test mail as before).

I think I can close now this discussion as resolved. Thank your for your time and work guys, very appreciated
